### PR TITLE
Throw API error if addon is not configured [OSF-4947]

### PR DESCRIPTION
**Pull request by samanehsan**
## Purpose

Going to the API endpoint `/v2/node_id/files/<provider>/` for an addon that is not configured on the project throws a 503 service unavailable error.

## Changes

Throw a 404 NotFound error.

## Side effects

None.

## Ticket
https://openscience.atlassian.net/browse/OSF-4947

### CODE REVIEW

**Comment by samanehsan** [comment]
The failing tests are also failing on develop, so this is ready for CR!

File: api/nodes/utils.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def get_file_object(node, path, provider, request):
             )
         return obj
 
+    if not node.has_addon(provider):
```
Comment (mfraezz) [inline api/nodes/utils.py]: This doesn't work if the addon is authorized but not configured.

screen shot 2016-05-02 at 09 55 41
screen shot 2016-05-02 at 09 55 54


You can check that with `node.get_addon(provider).configured`

**Comment by samanehsan** [comment]
Done making that change @mfraezz!

File: api/nodes/utils.py [hunk 2]
```diff
@@ -25,6 +25,9 @@ def get_file_object(node, path, provider, request):
             )
         return obj
 
+    if not node.get_addon(provider).configured:
```
Comment (mfraezz) [inline api/nodes/utils.py]: This can raise an `AttributeError` if `get_addon` returns `None`

Comment (samanehsan) [inline api/nodes/utils.py]: Yeah, I just realized that when my test failed :( This is fixed now!

**Comment by mfraezz** [comment]
LGTM. Dropping this in RTM

**Pull request merged**